### PR TITLE
Merge test and publish workflow

### PR DESCRIPTION
**Pull request by zyassine**
[JIRA NOOS-415](https://noosenergy.atlassian.net/browse/NOOS-415?atlOrigin=eyJpIjoiYjk2NzdiMjg0MDkxNGZiMjlmYjI3ZTQzNThkNDQyYTEiLCJwIjoiaiJ9)

### CODE REVIEW

**Review by pieeri — CHANGES_REQUESTED** [review]
Referring to CircleCI documentation on filters:
https://circleci.com/docs/configuration-reference/#tags

In relation to the project main config file, `config.yml`, the workflow isn't triggered by default on git tags. To run the workflow, a tag filter needs to be added. As a reference,
https://github.com/CircleCI-Public/Orb-Template/blob/main/.circleci/config.yml#L9

Another good working example for master/tag workflow switch,
https://github.com/noosenergy/noos-invoke/blob/master/.circleci/config.yml#L112

**Comment by zyassine** [comment]
per the documentation, I added filters for all jobs

File: .circleci/test-deploy.yml [hunk 1]
```diff
@@ -129,7 +129,7 @@ jobs:
 # ------------------
 
 workflows:
-  test-publish:
+  test:
     jobs:
       - integration_test_python_lint_package:
           filters: *filters
```
Comment (pieeri) [inline .circleci/test-deploy.yml:135]: No filters on git tags required here ; only run workflow jobs on all branches.

File: .circleci/test-deploy.yml [hunk 2]
```diff
@@ -4,6 +4,14 @@ version: 2.1
 # https://circleci.com/docs/testing-orbs/
 # https://github.com/CircleCI-Public/Orb-Template/blob/main/.circleci/test-deploy.yml
 
+# ----------------
+# Filters
+# ----------------
+
+filters: &filters
```
Comment (pieeri) [inline .circleci/test-deploy.yml:11]: Block to be removed ; not required. After that, good to go! Well done!

**Review by pieeri — CHANGES_REQUESTED** [review]
Last breadcrumb left behind.

**Review by pieeri — APPROVED** [review]
LGTM! Thanks a lot!

**Pull request merged**